templates/zlogin

```diff
@@ -8,52 +8,38 @@
 #
 
 (
-  # Function to determine the need of a zcompile. If the .zwc file
-  # does not exist, or the base file is newer, we need to compile.
-  # These jobs are asynchronous, and will not impact the interactive shell
-  zcompare() {
-    if [[ -s ${1} && ( ! -s ${1}.zwc || ${1} -nt ${1}.zwc ) ]]; then
-      zcompile ${1}
-    fi
-  }
-
   local zim_mods=${ZIM_HOME}/modules
   setopt EXTENDED_GLOB
+  autoload -U zrecompile
 
   # zcompile the completion cache; siginificant speedup.
   for file in ${ZDOTDIR:-${HOME}}/.zcomp^(*.zwc)(.); do
-    zcompare ${file}
+    zrecompile -pq ${file}
   done
 
   # zcompile .zshrc
-  zcompare ${ZDOTDIR:-${HOME}}/.zshrc
-
-  # zcompile some light module init scripts
-  zcompare ${zim_mods}/git/init.zsh
-  zcompare ${zim_mods}/git-info/init.zsh
-  zcompare ${zim_mods}/utility/init.zsh
-  zcompare ${zim_mods}/pacman/init.zsh
-  zcompare ${zim_mods}/spectrum/init.zsh
-  zcompare ${zim_mods}/completion/init.zsh
-  zcompare ${zim_mods}/fasd/init.zsh
-
-  # zcompile all .zsh files in the custom module
-  for file in ${zim_mods}/custom/**/^(README.md|*.zwc)(.); do
-    zcompare ${file}
+  zrecompile -pq ${ZDOTDIR:-${HOME}}/.zshrc
+
+  # zcompile all module init scripts
+  for file in ${zim_mods}/*/init.zsh; do
+    zrecompile -pq ${file}
   done
 
   # zcompile all autoloaded functions
-  for file in ${zim_mods}/**/functions/^(*.zwc)(.); do
-    zcompare ${file}
+  zrecompile -pq ${ZIM_HOME}/functions ${zim_mods}/*/functions/^([_.]*|prompt_*_setup|README*|*.zwc)(.)
+
```

Comment:
Now I see an issue with this approach: it will make available all module functions, even for the modules that are not loaded.

Reply:
perhaps wrap this in a for-loop that iterates over `zmodules` instead of wildcarding the path after `${zim_mods}`?

Reply:
the other downside to this, is when changes are made to the config file, we'll have to `source .zlogin` to regen everything; they will no longer be a part of fpath (should probably make a zmanage convenience function for this; perhaps "rebuild" or "compile"?)

Reply:
Only including `zmodules` fixes this. And `zrecompile` is smart enough to recompile the digest when the list of files changes. 

I'd like to apply the same approach to the init files, and only compile those for the enabled `zmodules`.

I don't get the downside you mentioned. In my understanding, in the current implementation a `source ~/.zlogin` is already needed to load new modules, and only starting a new shell would work to "not load" any removed modules. In the new solution, `fpath` never changes, not even when `zmodule` changes, so maybe this is even a upside...  :- )

Reply:
So right now, if we compile all of the modules into a single digest and add that digest to the fpath, we've enabled all of the modules. The `zmodules` content isn't considered; they're all enabled.

To be clear, I think we should only compile functions and init scripts for modules enabled in the `zmodules` array.